linkerd/app/outbound/src/tcp/connect.rs

```diff
@@ -50,9 +51,12 @@ impl<C> Outbound<C> {
             + svc::Param<Option<http::AuthorityOverride>>
             + svc::Param<Option<SessionProtocol>>
             + svc::Param<transport::labels::Key>,
-        C: svc::Service<Connect, Error = io::Error> + Clone + Send + 'static,
-        C::Response: tls::HasNegotiatedProtocol,
-        C::Response: io::AsyncRead + io::AsyncWrite + Send + Unpin + 'static,
+        C: svc::MakeConnection<Connect, Metadata = Local<ClientAddr>, Error = io::Error>
+            + Clone
+            + Send
+            + 'static,
```

Comment:
style nit, TIOLI: consider
```suggestion
        C: svc::MakeConnection<Connect, Metadata = Local<ClientAddr>, Error = io::Error>
        C: Clone + Send + 'static,
```